### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by auroter**
This is a quick refactor.  Here's what was changed:

1) Environment related functions and constants were moved into Environments.cs
2) Duplicate Environment variable was removed from StringXYZ class (it is already in ApiClient like the static headers)
3) Naming conventions updated
4) Actually use default env enum instead of hardcoding its resulting string in the constructor
5) Gave CardHolder a constructor so that it can be easily passed into SubmitCard(new("cardholder name"))
6) C style curly braces.

### CODE REVIEW

File: Runtime/Environments.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+namespace StringSDK
+{
+    public class Environment
+        {
+        private const string URL_API_PROD = "https://api.string-api.xyz";
```
Comment (frostbournesb) [inline Runtime/Environments.cs]: Why reverse the order of the words?

File: Runtime/Environments.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        private const string URL_API_SANDBOX = "https://api.sandbox.string-api.xyz";
+        private const string URL_API_DEV = "https://string-api.dev.string-api.xyz";
+        // TODO: Remove exposed local env before product launch
+        private const string URL_API_LOCAL = "http://localhost:5555";
+
+        public enum Types
```
Comment (frostbournesb) [inline Runtime/Environments.cs]: This should be singular
https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-classes-structs-and-interfaces#naming-enumerations
Also, is it possible to expose this to the user as StringSDK.EnvironmentType.xx?

File: Runtime/Environments.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            {
+                Types.PROD => URL_API_PROD,
+                Types.SANDBOX => URL_API_SANDBOX,
+                Types.DEV => URL_API_DEV,
+                Types.LOCAL => URL_API_LOCAL,
+                _ => ToUrl(DEFAULT),
+            };
+        }
+
+        public static Types ToType(string url)
```
Comment (frostbournesb) [inline Runtime/Environments.cs]: Why another switch table over setting `static EnvironmentType environment;` on the property mutator?

File: Runtime/API.cs [hunk 4]
```diff
@@ -8,51 +8,17 @@
...
-                EnvironmentType.SANDBOX => Constants.SANDBOX_API_URL,
-                EnvironmentType.DEV => Constants.DEV_API_URL,
-                EnvironmentType.LOCAL => Constants.LOCAL_API_URL,
-                _ => Constants.DEFAULT_ENV,
-            };
-        }
-
-        public static EnvironmentType Environment
+        // Environment
+        public static Environment.Types ENV
```
Comment (frostbournesb) [inline Runtime/API.cs]: Shouldn't this be `Env`, as properties should be PascalCase instead of CONSTANT_CASE

File: Runtime/StringEvents.cs [hunk 5]
```diff
@@ -11,6 +11,8 @@ public class Cardholder
 {
     // cardholder's full name e.g John Doe. 
     public string name;
+
+    public Cardholder(string name) {this.name = name;}
```
Comment (frostbournesb) [inline Runtime/StringEvents.cs:15]: Other constructors in this file don't use single-line shorthand
```suggestion
    public Cardholder(string name)
    {
        this.name = name;
    }
```

**Review by frostbournesb — COMMENTED** [review]
Another refactor you can do is have apiClient.BaseUrl use autoimplemented `get; set;`
In fact we don't need a property at all and can just do internal string baseUrl and set that

**Review by saito-sv — APPROVED** [review]
LGTM

**Review by saito-sv — APPROVED** [review]

**Review by frostbournesb — APPROVED** [review]

**Pull request merged**